### PR TITLE
Adding CLI Test Job for GP7 for dev pipeline

### PR DESCRIPTION
**Pull request by hpandeycodeit**
- In GP7, gpscp is replaced by gpsync. Introduced a variable (GP_SCP_CMD) to specify the appropriate command based on GP version.
- Replaced the hardcoded pxf home (/usr/local/pxf-gp6) with the variable PXF_HOME.
- Added variables to the Makefile (GP7_CLI) and dev pipeline template (gp7_cli), when set to true, this will introduce PXF_CLI tests for GP7.
- Ran yamllint and shellcheck to confirm there are no issues with the new changes.
- For ccp, use PLATFORM value (rocky8-gpdb7) as the corresponding VM Image comes preinstalled with all GPDB7 runtime dependencies.

Note: See Jira details for dev-pipeline supporting this PR.

### CODE REVIEW

File: concourse/scripts/install_pxf.bash [hunk 1]
```diff
@@ -12,6 +12,14 @@ if [[ ${PXF_COMPONENT} == "true" ]]; then
 else
 	PXF_HOME=${GPHOME}/pxf
 fi
+
+# gpscp command is replaced with gpsync in GP7
+if [ "${GP_VER}" -lt 7 ]; then
```
Comment (denalex) [inline concourse/scripts/install_pxf.bash:17]:
```suggestion
if [ ${GP_VER} -lt 7 ]; then
```
otherwise you're comparing a string to a number ?

**Review by denalex — COMMENTED** [review]
Also should we change the title of the PR / commit message from "Adding CLI Test Job for GP7" to "Adding CLI Test Job for GP7 for dev pipeline" ?

File: concourse/scripts/install_pxf.bash (hunk 1 above)
Comment (bradfordb-vmware) [reply concourse/scripts/install_pxf.bash:17]: According to `man 1 test`, `INTEGER1 -lt INTEGER2` tests if `INTEGER1` is less than `INTEGER2`. If you omit the double-quotes, then shellcheck warns about globbing and word splitting. The quotes don't cause bash to treat the value as an string.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 2]
```diff
@@ -83,7 +83,7 @@ resource_types:
     password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
 {% endif %}
 
-{% if multinode %}
+{% if multinode or gp7_cli %}
```
[the remark this answers is not in the document]
Comment (bradfordb-vmware) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:86]: The CCP resource (`name: terraform`) is still using the `terraform` resource type so its still only the IPA cluster that uses Terraform 0.14.10.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 3]
```diff
@@ -618,6 +617,82 @@ jobs:
...
+    - do:
+      - put: terraform_gpdb
+        resource: terraform
+        params:
+          action: create
+          delete_on_failure: true
+          generate_random_name: true
+          terraform_source: ccp_src/google/
+          vars:
+            PLATFORM: rocky8-gpdb7
```
Comment (ashuka24) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:650]: in all the other locations, we only use the general platform `PLATFORM: rocky8`, is there a reason we are now specifying the gpdb version as well?

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 4]
```diff
@@ -618,6 +617,82 @@ jobs:
...
+    on_success:
+      in_parallel:
+        steps:
+        - *destroy_gpdb_cluster
+    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
+    file: pxf_src/concourse/tasks/test_pxf_cli.yml
+{% if slack_notification %}
+    <<: *slack_alert
+{% endif %}
+{% endif %}
```
Comment (ashuka24) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:694]: the beginning of this if is relatively far up, can we comment that this is the end if for the `if gp7-cli`

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml (hunk 3 above)
Comment (hpandeycodeit) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:650]: `rocky8-gpdb7` is provided by releng. It contains the needed runtime dependencies for GP7. Most recently this was updated to include python3 dependencies, which are no longer provided with GP7 rpm.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 5]
```diff
@@ -618,6 +617,82 @@ jobs:
...
+      PXF_JVM_OPTS: ((pxf-jvm-opts))
+  - task: Test PXF CLI
+    on_success:
+      in_parallel:
+        steps:
+        - *destroy_gpdb_cluster
+    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
+    file: pxf_src/concourse/tasks/test_pxf_cli.yml
+{% if slack_notification %}
+    <<: *slack_alert
```
[the remark this answers is not in the document]
Comment (hpandeycodeit) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:692]: Following existing slack alert patterns which are present in the majority of test jobs. There is slight inconsistency in non-extended tests usage of the slack alerts.

File: concourse/scripts/install_pxf.bash (hunk 1 above)
Comment (hpandeycodeit) [reply concourse/scripts/install_pxf.bash:17]: @denalex for your reference, quotes are fine here. We are concerned about the shellcheck issue that Bradford mentioned (shellcheck warns about globbing and word splitting)
Also, we have confirmed quoting has no impact on the numeric comparison. Please refer to this link : https://tldp.org/LDP/abs/html/comparison-ops.html  as it has examples of quoted numeric variables.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (edespino) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:86]: @denalex nice catch and @bradfordb-vmware good input. Unlike unused resources, Concourse does not warn/error with unused resource_types (i.e. `terraform-0.14.10`) exist in a pipeline. The `terraform-0.14.10` resource_type is only used in the `terraform_ipa_hadoop` which is only enabled with the `MULTINODE` (pipeline: `multinode`) configuration.

File: concourse/Makefile [hunk 6]
```diff
@@ -116,6 +117,7 @@ set-dev-build-pipeline:
 		gs=$(GS) \
 		minio=$(MINIO) \
 		oel7=$(OEL7) \
+		gp7_cli=$(GP7_CLI) \
```
[the remark this answers is not in the document]
Comment (edespino) [reply concourse/Makefile:120]: @denalex Thank you for your observation and question - The current pattern we use for the dev test configuration options doesn't provide an indication that an option is only applicable to a specific GP version. The CLI test is explicitly being extracted from within the *IMPERS* GP 6 jobs for GP7 use. We are attempting to provide developer friendly configuration option which indicates it is only applicable to a specific GP version (GP7_CLI). I would like to give this enhanced pattern a try with the intent that we can get some developer feedback during actual use and determine if it is a worthwhile to introduce in other applicable scenarios.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 7]
```diff
@@ -618,6 +617,81 @@ jobs:
...
+  - task: Test PXF CLI
+    on_success:
+      <<: *destroy_gpdb_cluster
+    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
+    file: pxf_src/concourse/tasks/test_pxf_cli.yml
+{% if slack_notification %}
+    <<: *slack_alert
+{% endif %}
+{% endif %}
+## ----------GP7 PXF CLI tests END ----------
```
Comment (edespino) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:693]: We need to unset gp_ver at the end of the new GP 7 CLI block.

`{% set gp_ver = None %}`

File: concourse/Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (hpandeycodeit) [reply concourse/Makefile:120]: As discussed during the meeting, we are keeping it  as `GP7_CLI`.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hpandeycodeit) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml]: Thanks for pointing it out! I have tested it in the pipeline and this resource type `terraform-0.14.10` is not needed for this job. Updated the template file as well.

**Review by edespino — APPROVED** [review]
+1

**Review by yimingli-vmware — APPROVED** [review]

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 8]
```diff
@@ -618,6 +617,81 @@ jobs:
 {% endfor %}
 {% set gp_ver = None %}
 
+## ---------- GP7 PXF tests ----------
```
Comment (ashuka24) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:620]: the only test in this section is the CLI and we have other GP7 related test jobs above. should we update this comment?

**Review by ashuka24 — APPROVED** [review]

File: concourse/scripts/install_pxf.bash (hunk 1 above)
Comment (denalex) [reply concourse/scripts/install_pxf.bash:17]: then at least put 7 in quotes ?

**Pull request merged**